### PR TITLE
Fix #323 - Dynamic dropdown

### LINKED ISSUE #323: Status field is dependent on state field but it not responding to changes
#### Issue

he status field in the Cases module is showing all values no matter what value is selected in the State field.

#### Expected Behavior

“New” and “In progress” Status drop-down values will be displayed when the State field is set to “Open”.
“Resolved” and “Cancelled” Status field drop-down values will be displayed when the State field is set to “Closed”.

#### Actual Behavior

The status field in the Cases module is showing all values no matter what value is selected in the State field.

#### Possible Fix

checkAndInitAsDynamicEnum() method need to be added to dropdownenum.component. Because the method checks the values depending on a dependant field.

#### Steps to Reproduce

1. In base-field.manifest.ts file change following:
   From :
    'enum.detail': EnumDetailFieldComponent,
    'enum.edit': EnumEditFieldComponent,
   To: 
   'enum.detail': DropdownEnumDetailFieldComponent,
    'enum.edit': DropdownEnumEditFieldComponent,
2. In editview, select State and check Status field. You ll see all values.
3.Select another value in State, and still you will see all the values

#### Context

Status needs to be dynamic dependent on State field. This is a low priority.

#### Your Environment

* SuiteCRM Version used: 8.3.0
* Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)): Chrome Version 115.0.5790.171 (64-bit)
* Environment name and version (e.g. MySQL, PHP 7): Wamp Server, PHP 7.4.33, MariaDB 10.4.12
* Operating System and version (e.g Ubuntu 16.04): Windows 11

### PR DESCRIPTION
**Pull request by yunusyerli1**
## Description

Make dropdown fields dynamic#323 

## Motivation and Context

The status field in the Cases module is showing all values no matter what value is selected in the State field.

## How To Test This

In base-field.manifest.ts file change following:
From :
'enum.detail': EnumDetailFieldComponent,
'enum.edit': EnumEditFieldComponent,
To:
'enum.detail': DropdownEnumDetailFieldComponent,
'enum.edit': DropdownEnumEditFieldComponent,
In editview, select State and check Status field. You ll see all values.
Select another value in State, and still you will see all the values

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [x] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ ] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

### CODE REVIEW

**Comment by serhiisamko091184** [comment]
Hey @yunusyerli1,
thanks a lot for current contribution! 

Regards,
Serhii

File: core/app/core/src/lib/fields/base/base-field.component.ts [hunk 1]
```diff
@@ -73,7 +79,12 @@ export class BaseFieldComponent implements FieldComponentInterface, OnInit, OnDe
         const fieldKeys = (this.record.fields && Object.keys(this.record.fields)) || [];
         if (fieldKeys.length > 1) {
             this.calculateDependentFields(fieldKeys);
-            this.field.previousValue = this.field.value;
+            this.previousValue = deepClone({
+                value: this.field.value,
+                valueList: this.field.valueList,
+                valueObject: this.field.valueObject,
+                forceNotEqual: 'forceNotEqual',
```
Comment (clemente-raposo) [inline core/app/core/src/lib/fields/base/base-field.component.ts:86]: Hi @yunusyerli1, what is `forceNotEqual` used for?

**Comment by lukio** [comment]
This PR fixes **Make dropdown fields dynamic** #323.
Also, this PR is need it by PR #400 to works.

**Comment by johnM2401** [comment]
Hey @yunusyerli1! 
Thank you for your PR!

We weren't able to bring in your PR as we made several changes to the enum, multienum and relate fields for edit and filter modes which required us to do extra changes additional to the ones you suggested

However, this PR and your contributions are still greatly appreciated.

I will close this off as the issue has been resolved by the changes to enum-type fields in the latest release. (8.6.0)

**Pull request closed without merging**